src/statement.rs

```diff
@@ -1165,28 +1320,40 @@ mod test {
         .unwrap();
         // existing collection:
         let data = vec![1, 2, 3];
-        db.query_row("SELECT ?1, ?2, ?3", &data, |row| row.get::<_, u8>(0))
-            .unwrap();
-        db.query_row("SELECT ?1, ?2, ?3", data.as_slice(), |row| {
+        db.query_row("SELECT ?1, ?2, ?3", params_from_iter(&data), |row| {
```

Comment:
Oh no !

Reply:
Do you think this is that bad? The case where you actually are passing in a list that isn't either `&[&dyn ToSql]` is pretty rare IME — the only cases where you actually need to do that are cases where you're dynamically generating SQL I think

Reply:
Yes I know ! `params!` macro (and all the rest: `IntoIterator`, ...) was introduced only because of this specific case...
But that is fine.